ballerina-shell/modules/shell-cli/src/main/java/io/ballerina/shell/cli/jline/validator/ModuleMemberValidator.java

```diff
@@ -52,6 +53,16 @@ public boolean evaluate(String source) {
         ModulePartNode node = tree.rootNode();
         // Ignore code segments including imports
         Node parsedNode = NodeParser.parseModuleMemberDeclaration(source);
+        if (node.members().size() > 1) {
+            Node lastNode = node.members().get(node.members().size() - 1);
+            // Avoid false ModuleVariableDeclarations and special case functions
+            // Sample testcase : if (x == y) { x = x + 1; x = x + 1;
+            if (lastNode.kind() == SyntaxKind.FUNCTION_DEFINITION) {
```

Comment:
Shouldn't we consider these scenarios as well?
`"int a = 10; if (a>5) {} else {"`

Reply:
Since we are considering the `Module Level` above example wont execute. If we are considering above case we need to update the shell parser side as well.